src/Microsoft.AspNetCore.Mvc.Core/ControllerContext.cs

```diff
@@ -52,18 +52,18 @@ public new ControllerActionDescriptor ActionDescriptor
         }
 
         /// <summary>
-        /// Gets or sets the list of <see cref="IValueProvider"/> instances for the current request.
+        /// Gets or sets the list of <see cref="IValueProviderFactory"/> instances for the current request.
         /// </summary>
-        public virtual IList<IValueProvider> ValueProviders
+        public virtual IList<IValueProviderFactory> ValueProviderFactories
```

Comment:
In what scenarios could this list be changed? That is, could the property instead be an `IReadOnlyList<IValueProviderFactory>`?